tools/run_tests/helper_scripts/build_cxx.bat

```diff
@@ -38,14 +38,13 @@ If "%GRPC_CMAKE_GENERATOR%" == "Ninja" (
 
   @rem Select MSVC compiler (cl.exe) explicitly to make sure we don't end up gcc from mingw or cygwin
   @rem (both are on path in kokoro win workers)
-  cmake -G "%GRPC_CMAKE_GENERATOR%" -DCMAKE_C_COMPILER="cl.exe" -DCMAKE_CXX_COMPILER="cl.exe" -DgRPC_BUILD_TESTS=ON -DCMAKE_BUILD_TYPE="%MSBUILD_CONFIG%" %* ../.. || goto :error
+  cmake -G "%GRPC_CMAKE_GENERATOR%" -DCMAKE_VS_PLATFORM_TOOLSET_HOST_ARCHITECTURE=x64 -DCMAKE_C_COMPILER="cl.exe" -DCMAKE_CXX_COMPILER="cl.exe" -DgRPC_BUILD_TESTS=ON -DCMAKE_BUILD_TYPE="%MSBUILD_CONFIG%" %* ../.. || goto :error
```

Comment:
I don't think just using -DCMAKE_VS_PLATFORM_TOOLSET_HOST_ARCHITECTURE=x64 for ninja works. The option is only supported by the visual studio generator (i.e. `-G Visual Studio XYZ` not for `-G Ninja` like here. If this gets accepted, I think it will simply get ignored by the ninja build. Note that with Ninja generator, cmake doesn't really pick the toolset for you. It just realies on cl.exe compiler to be on PATH,  which is done by the GRPC_BUILD_ACTIVATE_VS_TOOLS block above.

Reply:
I think I found what you need:
https://learn.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=msvc-170#developer_command_file_locations

You'll need to modify the  GRPC_BUILD_ACTIVATE_VS_TOOLS block to pass the right arguments to vcvarsall.bat or use a different .bat file (vcvarsamd64_x86.bat in this case?).

Reply:
Passing `x64_x86` instead of just `x86` to vcvarsall.bat (on line 31) should be enough here?

Reply:
Jan, how about having this change for msvc build first since it's been failing? We can follow up on ninja after that. Since I'm not familar with Ninja build setup, I'm not sure I can make a right change soon enough.